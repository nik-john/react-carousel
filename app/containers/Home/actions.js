/*
 *
 * Home actions
 *
 */

import {
  LOAD_IMAGES,
  LOAD_IMAGES_ERROR,
  LOAD_IMAGES_SUCCESS,
} from './constants';

export function loadImages() {
  return {
    type: LOAD_IMAGES,
  };
}
export function loadImagesSuccess(images) {
  return {
    type: LOAD_IMAGES_SUCCESS,
    images,
  };
}
export function loadImagesError(error) {
  return {
    type: LOAD_IMAGES_ERROR,
    error,
  };
}
