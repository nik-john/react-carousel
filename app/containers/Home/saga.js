import { call, put, takeLatest } from 'redux-saga/effects';
import request from 'utils/request';
import { LOAD_IMAGES } from 'containers/Home/constants';
import { loadImagesSuccess, loadImagesError } from 'containers/Home/actions';

/**
 *Pixabay API handler
 */
export function* getImages() {
  // const API_KEY = process.env.API_KEY;
  const API_KEY = '9338657-e7b2ac124243b90813e3ed097';
  const requestURL = `https://pixabay.com/api?q=cats&min_width=200&min_height=200&key=${API_KEY}`;
  try {
    const images = yield call(request, requestURL);
    yield put(loadImagesSuccess(images));
  } catch (err) {
    yield put(loadImagesError(err));
  }
}

// Individual exports for testing
export default function* defaultSaga() {
  // See example in containers/HomePage/saga.js
  yield takeLatest(LOAD_IMAGES, getImages);
}
