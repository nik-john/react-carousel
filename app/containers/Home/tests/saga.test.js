import { put } from 'redux-saga/effects';
import { getImages } from '../saga';
import { loadImagesSuccess, loadImagesError } from '../actions';

/**
 * Test  sagas
 */

/* eslint-disable redux-saga/yield-effects */
// import { take, call, put, select } from 'redux-saga/effects';
// import { defaultSaga } from '../saga';

// const generator = defaultSaga();

describe('defaultSaga Saga', () => {
  let getImagesGenerator;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getImagesGenerator = getImages();

    const callDescriptor = getImagesGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });
  it('should dispatch the loadImagesSuccess action if it requests the data successfully', () => {
    const response = [{
      name: 'First image',
    }, {
      name: 'Second image',
    }];
    const putDescriptor = getImagesGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(loadImagesSuccess(response)));
  });

  it('should call the repoLoadingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getImagesGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(loadImagesError(response)));
  });
});
