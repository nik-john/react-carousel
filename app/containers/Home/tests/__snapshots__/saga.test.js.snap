// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultSaga Saga should call the repoLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://pixabay.com/api?q=cats&min_width=200&min_height=200&key=9338657-e7b2ac124243b90813e3ed097",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`defaultSaga Saga should dispatch the loadImagesSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "https://pixabay.com/api?q=cats&min_width=200&min_height=200&key=9338657-e7b2ac124243b90813e3ed097",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
